### PR TITLE
[rfxcom] Add support for receiving RAW messages

### PR DESCRIPTION
**Pull request by Jamstah**
This is a new feature in the Pro firmwares that provides the real raw RF
pulse lengths as shorts. Good for being able to parrot switches that aren't
otherwise supported, once we add the tx support as well.

Signed-off-by: James Hewitt <james.hewitt@uk.ibm.com>

### CODE REVIEW

File: bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    public void setDeviceId(String deviceId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void convertFromState(String channelId, Type type) throws RFXComUnsupportedChannelException {
+        switch (channelId) {
+            case CHANNEL_RAW_MESSAGE:
+                if (type instanceof StringType) {
+                    // TODO: Check the raw message for validity (length, no more than 124 shorts, multiple of 4 bytes in
```
Comment (martinvw) [inline bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java:160]: There is a TODO, what should be done and can it be completed before merging the PR?

File: bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    // payload)
+                    throw new RFXComUnsupportedChannelException("Channel " + channelId + " inot yet implemented");
+                } else {
+                    throw new RFXComUnsupportedChannelException("Channel " + channelId + " does not accept " + type);
+                }
+
+            case CHANNEL_RAW_PAYLOAD:
+                if (type instanceof StringType) {
+                    // TODO: Check the payload for validity (no more than 124 shorts, multiple of 4 bytes
```
Comment (martinvw) [inline bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java:169]: There is a TODO, what should be done and can it be completed before merging the PR?

File: bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComBindingConstants.java [hunk 3]
```diff
@@ -145,6 +145,7 @@
     private static final ThingTypeUID THING_TYPE_POWER = new ThingTypeUID(BINDING_ID, "power");
     private static final ThingTypeUID THING_TYPE_RADIATOR1 = new ThingTypeUID(BINDING_ID, "radiator1");
     private static final ThingTypeUID THING_TYPE_RAIN = new ThingTypeUID(BINDING_ID, "rain");
+    private static final ThingTypeUID THING_TYPE_RAW = new ThingTypeUID(BINDING_ID, "raw");
```
Comment (martinvw) [inline bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComBindingConstants.java:148]: Just out of curiosity what is the difference with `undecoded`

Comment (Jamstah) [reply bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComBindingConstants.java:148]: UNDECODED is where the RFXCOM understands the type, but just hasn't broken it down into values. RAW is where the RFXCOM has received an RF message and hasn't tried to do anything to it at all, so the values are the ms pulse timings that would need to be processed into stable bytes before they could even be decoded.

File: bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java (hunk 1 above)
Comment (Jamstah) [reply bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java:160]: This is really a TODO for when I implement tx, which is next. There is value in merging rx.

File: bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java (hunk 2 above)
Comment (Jamstah) [reply bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComRawMessage.java:169]: This is really a TODO for when I implement tx, which is next. There is value in merging rx.

**Comment by martinvw** [comment]
Does it make to wait with merging until the tx support is added?

**Comment by Jamstah** [comment]
I would merge it, it adds value as is.

Not sure when exactly I'll find time to do the tx side.

**Review by martinvw — APPROVED** [review]

**Pull request merged**